miasm2/arch/aarch64/sem.py

```diff
@@ -127,6 +127,28 @@ def extend_arg(dst, arg):
     if op == 'SXTW':
         base = reg.signExtend(dst.size)
         op = "<<"
+    elif op == "SXTB":
+        base = reg.signExtend(dst.size)
+        op = "<<"
```

Comment:
Hi!

Ok there is an error in the semantic description here, but I can't blame you: the original SXTW was wrong :smile: 
As the documentation says the steps are:
- first truncate (Byte, Word)
- extend (signed /unsigned)
- shift (optionnal)

So the SXTW (and others !) should be something like:
```python
     if op == 'SXTW':
         base = reg[:16].signExtend(dst.size)
         op = "<<"
```

Do you agree with this?

Reply:
I totally agree with it. I am guilty of skipping this info too. I just assumed SXTW was correctly written. I will update the patch and submit again.